knative-operator/cmd/manager/main.go

```diff
@@ -7,55 +7,49 @@ import (
 	"fmt"
 	"os"
 
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/controller"
-	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook"
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/knativeeventing"
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/knativekafka"
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/knativeserving"
 	"github.com/operator-framework/operator-sdk/pkg/leader"
-	"github.com/operator-framework/operator-sdk/pkg/log/zap"
-	"github.com/operator-framework/operator-sdk/pkg/restmapper"
 	"github.com/spf13/pflag"
-
-	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	zapr "sigs.k8s.io/controller-runtime/pkg/log/zap"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 // Change below variables to serve metrics on different host or port.
 var (
 	metricsHost               = "0.0.0.0"
 	metricsPort         int32 = 8383
 	operatorMetricsPort int32 = 8686
+	log                       = logf.Log.WithName("cmd")
 )
-var log = logf.Log.WithName("cmd")
 
-func main() {
-	// Add the zap logger flag set to the CLI. The flag set must
-	// be added before calling pflag.Parse().
-	pflag.CommandLine.AddFlagSet(zap.FlagSet())
+func init() {
+	prodConf := zap.NewProductionEncoderConfig()
```

Comment:
Do we need a `NewDevelopmentConfig()` to make timestamps more readable etc for the developer enforcing a rule about timestamp format?
Also in production I think its useful either to enforce `UTC` or use local time + `offset from UTC`.

Reply:
Not parsing, didn't I configure the timestamp format below this line? The current config should be equivalent to what it was before I think. If we want additional changes, I'd do them separately and to all of our pods to be the same.

Reply:
I think we're doing what you're asking already. Here's an example log of this build

```
{"level":"info","ts":"2020-10-28T16:29:05.323Z","logger":"knative.openshift.health dashboard.mf","msg":"Parsing manifest"}
```

We log humanly readable (ISO8601) and locked at UTC (signified by the timestamp ending in a `Z`)